src/Sajari/proto/doc.php

```diff
@@ -0,0 +1,909 @@
+<?php
+// DO NOT EDIT! Generated by Protobuf-PHP protoc plugin 1.0
+// Source: doc.proto
+//   Date: 2016-05-20 03:33:14
+
+namespace sajari\engine\store\doc {
+
+  class XEmpty extends \DrSlump\Protobuf\Message {
```

Comment:
@tbillington re your autoloading problem: have you checked if the protobuf generator can be configured to generate the files for PSR-0?

If it can't, then another option that might work but you need to verify: I think you can specify in composer.json that when it finds a namespace `sajari\engine\store\doc` (for example), to look in directory `src/Sajari/proto/` (or maybe look at the doc.php file in there), and then it should find the class.